umap/static/umap/js/modules/sync/engine.js

```diff
@@ -0,0 +1,106 @@
+import { WebSocketTransport } from './websocket.js'
+import {
+  MapUpdater,
+  MarkerUpdater,
+  PolygonUpdater,
+  PolylineUpdater,
+  DataLayerUpdater,
+} from './updaters.js'
+
+export class SyncEngine {
+  constructor(map) {
+    this.map = map
+    this.receiver = new MessagesDispatcher(this.map)
+
+    this._initialize()
+  }
+  _initialize() {
+    this.transport = undefined
+    const noop = () => undefined
```

Comment:
```suggestion
    const noop = () => {}
```

This is a pattern that I'm more used to see in the JS community.